src/Streamly/Internal/LZ4.hs

```diff
@@ -154,6 +162,90 @@ cIntToI32 = coerce
 i32ToCInt :: Int32 -> CInt
 i32ToCInt = coerce
 
+--------------------------------------------------------------------------------
+-- Configuration
+--------------------------------------------------------------------------------
```

Comment:
We can move the config stuff to a separate file. Also, we should be careful about the `TypeFamilies` extension, it can lead to perf regression. We should either use `NoMonoLocalBinds` along with it or use it only in specific files instead of enabling for everything.